### ISSUE TITLE
PgLock.new(name: "lock").lock! false exception thrown in v0.2.0

### ISSUE THREAD

**Issue by awwafe** [heroku/pg_lock#14]
With the new return value from version 0.2.0, the `:lock!` method is now throwing a PgLock::UnableToLockError exception if the return value in false (even if PgLock was able to obtain a lock and run the content).

Can you make it so that `:lock!` throws a PgLock::UnableToLockError exception only if PgLock was unable to obtain the lock, or otherwise return the value correctly, even if it returns false/nil ?

```
> PgLock.new(name: "hello").lock! { puts "123" }
123
PgLock::UnableToLockError: Was unable to aquire a lock "hello" after 3 attempts
```

**Comment by schneems** [comment]
Oh no, sorry for the regression. Do you have a patch ready to go to fix it?
I can try to get to it soon but I don’t know when I’ll be able. Will also
need a test to prevent future regression.

-- 
Richard Schneeman
https://www.schneems.com

**Comment by awwafe** [comment]
Thanks for the quick reply.
I'll see what I can do about a patch when I can get to it also.

**Comment by schneems** [comment]
I'm not able to reproduce this behavior. I made a PR with a test but can't get it to fail https://github.com/heroku/pg_lock/pull/15

Can you let me know if you can get that to fail or alternatively can you provide me with an example app that reproduces the problem? https://www.codetriage.com/example_app

**Comment by awwafe** [comment]
[This](https://github.com/heroku/pg_lock/pull/15#pullrequestreview-325901903) comment is correct, if the code inside the lock! block returns a false or nil value, then the exception will be thrown.

**Comment by schneems** [comment]
Released 0.2.1 with my fix. Sorry about the regression!
